WordPress/src/main/java/org/wordpress/android/ui/stats/refresh/lists/sections/insights/usecases/AnnualSiteStatsUseCase.kt

```diff
@@ -42,57 +52,71 @@ class AnnualSiteStatsUseCase
-        items.add(
-                QuickScanItem(
-                        Column(
-                                string.stats_insights_year,
-                                lastYear.year
-                        ),
-                        Column(
-                                string.stats_insights_posts,
-                                lastYear.totalPosts.toFormattedString()
-                        )
-                )
-        )
-        items.add(
-                QuickScanItem(
-                        Column(
-                                string.stats_insights_total_comments,
-                                lastYear.totalComments.toFormattedString()
-                        ),
-                        Column(
-                                string.stats_insights_average_comments,
-                                lastYear.avgComments?.toFormattedString() ?: "0"
-                        )
-                )
-        )
-        items.add(
-                QuickScanItem(
-                        Column(
-                                string.stats_insights_total_likes,
-                                lastYear.totalLikes.toFormattedString()
-                        ),
-                        Column(
-                                string.stats_insights_average_likes,
-                                lastYear.avgLikes?.toFormattedString() ?: "0"
-                        )
-                )
-        )
-        items.add(
-                QuickScanItem(
-                        Column(
-                                string.stats_insights_total_words,
-                                lastYear.totalWords.toFormattedString()
-                        ),
-                        Column(
-                                string.stats_insights_average_words,
-                                lastYear.avgWords?.toFormattedString() ?: "0"
+
+        when (useCaseMode) {
+            BLOCK -> {
+                items.add(Title(R.string.stats_insights_this_year_site_stats))
+                items.addAll(annualStatsMapper.mapYearInBlock(domainModel.years.last()))
+                if (domainModel.years.size > 1) {
+                    items.add(
+                            Link(
+                                    text = R.string.stats_insights_view_more,
+                                    navigateAction = NavigationAction.create(this::onViewMoreClicked)
+                            )
+                    )
+                }
+            }
+            VIEW_ALL -> {
+                items.addAll(
+                        annualStatsMapper.mapYearInViewAll(
+                                domainModel.years.getOrNull(index) ?: domainModel.years.last()
                         )
                 )
-        )
+            }
+        }
         return items
     }
+
+    private fun onViewMoreClicked() {
+        navigateTo(NavigationTarget.ViewAnnualStats)
+    }
+
+    private fun yearToDate(year: String): Date {
```

Comment:
We might want to consider moving this method into DateUtils or DateTimeUtils, wdyt?

Reply:
this is a bit specific to the stats I think. We are using here the last day of the year as a Date that represents the whole year. I don't think it would make sense anywhere else.